### PR TITLE
feat: add trailingSlash support (#11)

### LINKED ISSUE #11: feat: Support `trailingSlash` option
Would like to see the [trailingSlash](https://docs.astro.build/en/reference/configuration-reference/#trailingslash) option applied to the generated links. When `trailingSlash='always'`, a `/` would be added, when `never | ignore`, the current behavior would remain since plugin should only be acting on `{.md,.mdx}` files (See related #12).

```markdown
Markdown: [My MD Link](./anotherfile.md)
HTML: <a href="anotherfile/">My MD Link</a>

Markdown: [My MD Link](./anotherfile.md#something)
HTML: <a href="anotherfile/#something">My MD Link</a>

Markdown: [My MD Link](./anotherfile.md?x=something)
HTML: <a href="anotherfile/?x=something">My MD Link</a>
```

Great plugin, solves a major pain point, thank you for your work!

Samples with proposed fixes in PR #21:
- `always`: <https://stackblitz.com/edit/github-pdzrjo-ptwjhv>
- `never`: <https://stackblitz.com/edit/github-pdzrjo-gizrtv>
- `ignore`: <https://stackblitz.com/edit/github-pdzrjo-yp6ehb>

---

**Edit:** Added note about plugin not being able to be used when `trailingSlash='always'`.

**Edit Take 2:** After further testing/findings, edited the OP entirely to focus only on trailing slash behavior for **files that exist** to avoid mixing issues related to directories

### PR DESCRIPTION
**Pull request by techfg**
Resolves #11 

I based this PR off of PR #17 as it made solutioning for supporting `trailingSlash` more straightforward to resolve than the prior code base would have (e.g., several issues in resolving the final url would have required several more modifications to the code than what is contained in this PR).  If the decision is made to not merge #17, I can make the modifications to this one based on existing code but hopefully you'll agree #17 is worth merging :)

Regarding supporting the `trailingSlash` option, I spent some time reviewing what other Astro packages are doing around `trailingSlash` support since there isn't any official documentation on how `trailingSlash` is supposed to work because from an core Astro perspective, it's really only a `dev server` config flag in terms of how it responds to requests.  Integrations, etc. have utilized it to offer rendering support in urls but its somewhat mixed in terms of how `always`, `never` and `ignore` are applied.  With that said, the approach I landed on is as follows:

1. `always` - Will ensure any transformed url **always** contains a trailing slash regardless of file path or custom slug
2. `never` - Will ensure any transformed url **never** contains a trailing slash regardless of file path or custom slug
3. `ignore` - Will ensure any transformed url **remains identical** trailing slash wise to the input of either the file path or custom slug (whichever was used).  

`always` and `never` are straightforward but `ignore` warrants an example.  Assuming pages of `post-1.md` and `post-2.md` in the root folder of a content collection `blog`, the following would be produced from the markdown of `page-1.md`:

- When `page-2.md` does not have a custom slug:
```markdown
`without trailing slash in the file path`
[Checkout Post 2](./post-2.md) -> `/blog/post-2`
`with trailing slash in the file path`
[Checkout Post 2](./post-2.md/) -> `/blog/post-2/`
```
- When `page-2.md` has a custom slug:
```markdown
`custom slug of 'second-post'`
[Checkout Post 2](./post-2.md) -> `/blog/second-post`
`custom slug of 'second-post/'`
[Checkout Post 2](./post-2.md) -> `/blog/second-post/`
```

In short, `ignore` is controlled by what's in the input and provides a way to include a trailing slash when desired regardless of whether or not the file path was resolved or a custom slug was provided.

Open to any feedback/suggestions, thank you!

### CODE REVIEW

**Comment by vernak2539** [comment]
After the merge of #17 this PR now needs some merge conflicts resolved 😅

**Comment by vernak2539** [comment]
After the merge of #17 this PR now needs some merge conflicts resolved 😅

**Comment by vernak2539** [comment]
Hey! There's been a lot of changes since this PR was opened, causing conflicts in the file. I did end up updating the tests, which, IMO, are much better and grouped a bit more consistently around functionality. Happy to review this when ready, as I think it would be beneficial functionality!

**Comment by techfg** [comment]
Yep, working on the merge now, was waiting till #17 merged :)

**Comment by techfg** [comment]
@vernak2539 - Merged!  FYI, saw your ternary change from the last PR and made another small tweak to use a const value for the set & compare, hope you don't mind :)

File: src/index.mjs [hunk 1]
```diff
@@ -118,11 +125,20 @@ function astroRehypeRelativeMarkdownLinks(options = {}) {
...
+      const resolvedUrl = [
+        collectionName === NO_COLLECTION_NAME
+          ? ""
+          : path.posix.sep + collectionName,
         resolvedSlug,
       ].join(path.posix.sep);
 
+      // slug of empty string ('') is a special case in Astro for root page (e.g., index.md) of a collection
+      let webPathFinal = applyTrailingSlash(
+        (frontmatterSlug === "" ? "/" : frontmatterSlug) || url,
```
Comment (vernak2539) [inline src/index.mjs:137]: What do you think about doing the same thing here that was done for `NO_COLLECTION_NAME`?

Comment (techfg) [reply src/index.mjs]: lol, I never did like the way I wrote that code where the collectionName would be `''` when `collectionPathMode` was `root` - the collection name is still the collection name, it's just whether or not we use it in the path :)  Just pushed a commit that I'm happier with and accomplishes both of our goals I think.  Lemme know!

Comment (vernak2539) [reply src/index.mjs]: Looks great! Thanks!
